### PR TITLE
sql: implement `SHOW BR JOBS <n:int>` and `CANCEL BR JOB <n:int>`

### PR DESCRIPTION
**Pull request by YuJuncen**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/43072 close 

Problem Summary:
This PR implemented two new SQLs.

### What is changed and how it works?
`SHOW BR JOB QUERY <n:int>` will display the origin SQL of the backup job. <del>(***WARNING:** the secret keys may not be redacted. So it requires `BACKUP_ADMIN or SUPER` role to be executed.*)</del>
`CANCEL BR JOB <n:int>` will cancel a running BR job. This just trivially cancel the context of the BR CLI for now.
This PR also exported the `ID` row of the backups to the result of `SHOW BACKUPS` and `SHOW RESTORES`, so you can reference it at the former SQLs.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Documentation

- [x] Affects user behaviors
- [x] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Added support for `SHOW BR JOBS <n:int>` and `CANCEL BR JOB <n:int>`.
```

### CODE REVIEW

**Review by Leavrth — APPROVED** [review]

File: errno/errcode.go [hunk 1]
```diff
@@ -1063,6 +1063,7 @@ const (
 	ErrLoadDataInvalidOperation            = 8171
 	ErrLoadDataLocalUnsupportedOption      = 8172
 	ErrLoadDataPreCheckFailed              = 8173
+  ErrBRJobNotFound                       = 8174
```
Comment (BornChanger) [inline errno/errcode.go:1066]: format

**Review by BornChanger — COMMENTED** [review]
Others lgtm

**Review by BornChanger — APPROVED** [review]

**Comment by YuJuncen** [comment]
Oops... This PR mistakenly brings https://github.com/pingcap/tidb/pull/43562 to master. I think it is moral to check what branch we are in before committing. 😢

**Pull request merged**